### DISCUSSION TITLE
Project relationship with website

### DISCUSSION THREAD

**Discussion by bmanga** [open-watcom/open-watcom-v2#322] [General]
I would like to know if the project is related in any way with the www.openwatcom.com website,
and if not, if we could gain access to it. It seems to have a lot of useful resources, and most importantly a bugzilla, which IMO would be better suited at keeping track of compiler bugs than github issues.

**Comment by ArmstrongJ** [comment]
I could not disagree with you more about using Bugzilla.  Aside from the openwatcom.com website being down about 50% of the time, using Bugzilla would mean bug reporters would need another account on the OW Bugzilla server.  GitHub issues work so well with GitHub too, including integration with pull requests, easy referencing in Markdown comments, etc.  I see no point in going back to Bugzilla.

Additionally, that server is provided by the people at Perforce.  I'm not sure they'd be keen on our using GitHub for all source control.

**Comment by bmanga** [comment]
My main problem with it is that you cannot file an issue for a specific component (and users cannot put labels on the issue without permissions). Hence someone with permissions needs to manually tag every new issue by type (c++ frontend bug , c library bug, codegen bug etc) and release version to keep them sorted. Bugzilla's interface lets you enforce the submission of these details.
I also don't see bug tracking and issue tracking systems as mutually exclusive (bugs are issues but not all issues are bugs). 
Anyways I understand the points you make, and if you feel that github issues are sufficient I can live with that. I would still suggest however some better categorization of the incoming issues, as I would like to log the c++ frontend bugs as they get found and be able to easily filter by them.

**Comment by jmalak** [comment]
We are very small group of people interested in next development of OpenWatcom.
It is fork of official OW version, because interest for rapid change of code on official site is zero.
There was many talks about it but it looks like the official site development is really dead.
We don't want to depend on any proprietary support (official OpenWatcom site is supported by Perforce company), therefore I select public GitHub service before a few years. We don't need any many and we are able migrate to similar Git public service if we will need without big effort.
I understand your concern about GitHub issues, but up to now it solved our requirements.
I don't know if public Bugzilla services are available somewhere.
If yes than we could discuse about it.

**Comment by jmalak** [comment]
Anyway put here proposal for labels of your interest and somebody with write access to repository will add it.

**Comment by bmanga** [comment]
Thanks for the info @jmalak ! 
I'd like at least a 'cxx11' and a 'cxx-frontend' label, so I can keep track of of the progress on that front :)

**Comment by jayschwa** [comment]
It seems like there's [active development on openwatcom.org](http://perforce.openwatcom.org:4000/@md=d&cd=//depot/openwatcom/&c=TI8@//depot/openwatcom/?ac=43&mx=50), including commits from @jmalak. What is the distinction between the two forks now? Are they converging?

**Comment by jmalak** [comment]
OW 2 on GitHub is my fork to continue development because on oficial site is no interest to my contribution to their OW 2 version, it is frozen for years. This site is handled by group of people which are intolerant. Development is closed to perforce tools (strictly centralised tool) that they can kick-off you any time from server and you have no access to source (they do it to me). It is not open development. Therefore I created this fork and moved OW sources to git and use public services on GitHub that anybody can use OW sources and contribute how want without any similar restriction or attack from some group of people which say what you should do and what not.

OW2 fork on GitHub diverges from openwatcom.org site sources because it is now native 64-bit version tools, openwatcom.org site version is only 32-bit binary, some of OW2 fork tools support 64-bit objects (librarian, linker) but official site not etc. I am trying to hold backward compatibility with OW1.9.
Main task for OW 2 development is to be able generate 64-bit code for Intel architecture, now most of tools are 64-bit binaries to be able run natively on 64-bit version Windows and Linux.
Source code is more portable to different host systems and multiple 64-bit compilers are supported.

**Comment by jmalak** [comment]
Main limitation of OW build system is WGML compiler (for documentation) which is available only in executable binary form for DOS and OS/2. The only live project on official OW site is recreation of sources for WGML tool. I do contribution to this sources time-to-time to be more portable and we can build it by various compilers and for 64-bit hosts, because I have still access to this source code.
I don't want to lost my time by parallel development of this tool. WGML tool is only one of tools which has some sense for OW2 fork.

**Comment by jayschwa** [comment]
@jmalak, so you are not making commits to openwatcom.org perforce? Why do some of the commits have your name? https://github.com/open-watcom/owp4v1copy/commits/master

Added: looks like we commented at the same time, and that you answered my question with the second comment.
